include/onnxruntime/core/providers/dml/dml_provider_factory.h

```diff
@@ -30,6 +30,23 @@
 extern "C" {
 #endif
 
+enum class OrtDmlPerformancePreference {
+  Default,
+  HighPerformance,  // Default
+  LowPower
+};
+
+enum class OrtDmlDeviceFilter {
```

Comment:
>OrtDmlDeviceFilter [](http://example.com/codeflow?start=11&length=18)

Should these be flags instead

GPU = 1
NPU = 2

We dont need None as that corresponds to GPU | NPU.

Correct?

Reply:
If we remove None how will the user specify GPU or NPU? By not passing a device filter at all?

Reply:
I added one for "Both" instead. Is this fine?

Reply:
If these are actually flags, not mutually exclusive identifiers, then please write them flag-like to make it clear to callers, and use the respective names rather than literal. e.g.

```
enum {
    Gpu = 0x00000001,
    Npu = 0x00000002,
    Both = Gpu | Npu,
};
```

(naming) If "Both" means Gpu *or* Npu, then it could be weird when we add more flags later (because both means two). So I'd rather just call it "GpuOrNpu"